src/main/java/org/jitsi/jigasi/JvbConference.java

```diff
@@ -1599,4 +1609,111 @@ public String toString()
                 + "]@"+ hashCode();
         }
     }
+
+    /**
+     * Sets the chatroom presence for the participant.
+     *
+     * @param muted <tt>true</tt> for presence as muted,
+     * false otherwise
+     */
+    void setChatRoomAudioMuted(boolean muted)
+    {
+        if (mucRoom != null)
+        {
+            AudioMutedExtension audioMutedExtension = new AudioMutedExtension();
+
+            audioMutedExtension.setAudioMuted(muted);
+
+            OperationSetJitsiMeetTools jitsiMeetTools
+                = xmppProvider.getOperationSet(OperationSetJitsiMeetTools.class);
+
+            jitsiMeetTools
+                .sendPresenceExtension(mucRoom, audioMutedExtension);
+
+        }
+    }
+
+    /**
+     * Request Jicofo on behalf of Jigasi to mute a participant.
+     *
+     * @param bMuted <tt>true</tt> if request is to mute audio,
+     * false otherwise
+     * @return <tt>true</tt> if request succeeded, false
+     * otherwise
+     */
+    public boolean requestAudioMute(boolean bMuted)
+    {
+        StanzaCollector collector = null;
+        try
+        {
+            String roomName = callContext.getRoomName();
+            if (!roomName.contains("@"))
+            {
+                // we check for optional muc service
+                String mucService
+                    = JigasiBundleActivator.getConfigurationService()
+                        .getString("org.jitsi.jigasi.MUC_SERVICE_ADDRESS", null);
+                if (!StringUtils.isNullOrEmpty(mucService))
+                {
+                    roomName = roomName + "@" + mucService;
+                }
+            }
+
+            String jidString = roomName + "/" + callContext
+                                                    .getCallResource()
+                                                    .getLocalpartOrNull()
+                                                    .toString();
+            Jid memberJid = JidCreate.from(jidString);
+            String roomJidString = roomName + "/" + focusResourceAddr;
+            Jid roomJid = JidCreate.from(roomJidString);
+            String fromJidString = JigasiBundleActivator.getConfigurationService()
+                                        .getString(JITSI_MEET_JIGASI_USER_ID);
+
+            fromJidString += "/" + callContext
+                                    .getCallResource()
+                                    .getLocalpartOrNull()
+                                    .toString();
+
+            Jid fromJid = JidCreate.from(fromJidString);
+
+            MuteIq muteIq = new MuteIq();
+            muteIq.setJid(memberJid);
+            muteIq.setMute(bMuted);
+            muteIq.setType(IQ.Type.set);
+            muteIq.setTo(roomJid);
+            // From is ignored, XMPPConnection changes the from unless
+            // FromMode is set to UNCHANGED, Actor can be used instead
+            muteIq.setFrom(fromJid);
+            muteIq.setActor(fromJid);
```

Comment:
In general, you don't need to set from. It is a responsibility of the xmpp server to set it when sending to other xmpp endpoints. This is because of security, you can spoof with different from, but the server knows the authentication details on the current connection and set them before sending the packets to other participants.
The actor also should not be mandatory. We added that to be able to say who muted someone else, but when you are muting yourself this is not important.
Have you tested it with removing these two lines and also removing the
```
private static final String JITSI_MEET_JIGASI_USER_ID
        = "org.jitsi.jigasi.xmpp.acc.USER_ID";
```
?

Reply:
Removed, works ok without.